core/src/main/java/bio/terra/pearl/core/service/export/EnrolleeExportService.java

```diff
@@ -107,15 +111,20 @@ public List<ModuleExportInfo> generateModuleInfos(ExportOptions exportOptions, U
     }
 
     protected List<ModuleExportInfo> generateSurveyModules(ExportOptions exportOptions, UUID studyEnvironmentId) throws Exception {
-        // for now, only worry about the surveys currently configured for the environment
-        List<StudyEnvironmentSurvey> latestConfiguredSurveys = studyEnvironmentSurveyService.findAllByStudyEnvIdWithSurvey(studyEnvironmentId);
-        latestConfiguredSurveys.sort(Comparator.comparing(StudyEnvironmentSurvey::getSurveyOrder));
+        // get all surveys that have ever been attached, including inactive
+        List<StudyEnvironmentSurvey> configuredSurveys = studyEnvironmentSurveyService.findAllByStudyEnvIdWithSurvey(studyEnvironmentId, null);
+        configuredSurveys.sort(Comparator.comparing(StudyEnvironmentSurvey::getSurveyOrder));
         SurveyFormatter surveyFormatter = new SurveyFormatter(objectMapper);
         List<ModuleExportInfo> moduleExportInfos = new ArrayList<>();
-        for (StudyEnvironmentSurvey configuredSurvey : latestConfiguredSurveys) {
-            var surveyQuestionDefinitions = surveyQuestionDefinitionDao
-                    .findAllBySurveyId(configuredSurvey.getSurvey().getId());
-            moduleExportInfos.add(surveyFormatter.getModuleExportInfo(exportOptions, configuredSurvey.getSurvey(), surveyQuestionDefinitions));
+        Map<String, List<StudyEnvironmentSurvey>> surveysByStableId = configuredSurveys.stream().collect(
```

Comment:
Sorry, trying to understand what this is.. 
The stable ID identifies a survey, which may have multiple versions? And those versions are represented by `StudyEnvironmentSurvey`?

Reply:
Close!  A stableId identifies a survey, which might have multiple versions.  Some of those versions may have been attached to a StudyEnvironment at one time, via a StudyEnvironmentSurvey.